scripts/us_census/pep/population_estimate_by_race/preprocess.py

```diff
@@ -0,0 +1,936 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" A Script to process USA Census PEP monthly population data
+    from the datasets in provided local path.
+    Typical usage:
+    1. python3 preprocess.py
+    2. python3 preprocess.py -i input_data
+"""
+import os
+import re
+import sys
+
+import pandas as pd
+from absl import app
+from absl import flags
+from state_to_geoid import USSTATE_MAP
+from states_to_shortform import get_states
+
+pd.set_option("display.max_columns", None)
+
+_FLAGS = flags.FLAGS
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "input_data"
+
+if ("ip_path" in sys.argv and len(sys.argv) > 2):
+    default_input_path = sys.argv[2]
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+
+# Generating geoID by taking Geographical area as input
+def add_geo_id(df: pd.DataFrame) -> pd.DataFrame:
+    short_forms = get_states()
+    df['Short_Form'] = df['Geographic Area'].str.replace(
+        " ", "").apply(lambda row: short_forms.get(row, row))
+    df['geo_ID'] = df['Short_Form'].apply(
+        lambda rec: USSTATE_MAP.get(rec, pd.NA))
+    df = df.dropna(subset=['Geographic Area'])
+    return df
+
+
+def _clean_xls_file(df: pd.DataFrame, file: str) -> pd.DataFrame:
+    """
+    This method cleans the dataframe loaded from a xls file format.
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+     According to dataset, origin=0, sex=0
+     considered as they hold the total value.
+    """
+    #index=0
+    if "2020" in file:
+        df.drop(df[df['ORIGIN'] != 0].index, inplace=True)
+        df.drop(df[df['SEX'] != 0].index, inplace=True)
+        df = df.drop(['STATE','DIVISION','SUMLEV','SEX','ORIGIN'\
+            ,'AGE','REGION','ESTIMATESBASE2010','CENSUS2010POP'\
+            ,'POPESTIMATE042020'],axis=1)
+    else:
+        df = df.drop(['STATE','DIVISION','REGION','ESTIMATESBASE2000',\
+            'CENSUS2010POP','POPESTIMATE2010'],axis=1)
+    df = df.groupby(['NAME', 'RACE']).sum().transpose().stack(0).reset_index()
+    if "2020" in file:
+        df[0] = df[1] + df[2] + df[3] + df[4] + df[5] + df[6]
+    df['0'] = df[0]
+    df['1'] = df[1]
+    df['2'] = df[2]
+    df['3'] = df[3]
+    df['4'] = df[4]
+    df['5'] = df[5]
+    df['6'] = df[6]
```

Comment:
Can column names be defined here itslef.

Reply:
Renamed it.